src/Core/Framework/Adapter/Cache/Http/CacheResponseSubscriber.php

```diff
@@ -210,8 +212,19 @@ private function hasInvalidationState(array $cacheStates, array $states): bool
 
     private function buildCacheHash(Request $request, SalesChannelContext $context): string
     {
+        $ruleAreas = $this->ruleResolver->resolveRuleAreas($request, $context);
+
+        if (Feature::isActive('v6.8.0.0') || Feature::isActive('PERFORMANCE_TWEAKS') || Feature::isActive('CACHE_CONTEXT_HASH_RULES_OPTIMIZATION')) {
+            $ruleIds = $context->getRuleIdsByAreas($ruleAreas);
```

Comment:
Maybe move the `$ruleAreas` into the if condition, such that it is clear that it is only used for that, probably making the upgrade easier (and the code easier to understand).

```suggestion
        if (Feature::isActive('v6.8.0.0') || Feature::isActive('PERFORMANCE_TWEAKS') || Feature::isActive('CACHE_CONTEXT_HASH_RULES_OPTIMIZATION')) {
            $ruleAreas = $this->ruleResolver->resolveRuleAreas($request, $context);

            $ruleIds = $context->getRuleIdsByAreas($ruleAreas);
```

Reply:
@aragon999 thanks for the review!
i did it intentionally outside that if, so the extension already runs and plugins etc will already be called, the thought was that that would make the update easier as your plugin will be called in every case and not depending on the feature flag 🤔 

I could clarify that with a comment?

Reply:
Ah I see, I did not think about this, but you are right :-)

I think in that case it is fine :-)